lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsService.cs

```diff
@@ -113,5 +113,18 @@ public void OnCompleted()
         public void OnError(Exception error)
         {
         }
+
+        /// <summary>
+        /// Observer of IDriverMetrics.
+        /// When Driver metrics data is changed, this method will be called
+        /// </summary>
+        /// <param name="value"></param>
+        public void OnNext(IDriverMetrics value)
+        {
+            var set = new HashSet<KeyValuePair<string, string>>();
```

Comment:
Talking about uniqueness, you are right, it doesn't need to be a set. May be to use List.